.github/workflows/ci.yml

```diff
@@ -200,6 +203,7 @@ jobs:
           S3_BUCKET_ENDPOINT: "s3.us-east-2.amazonaws.com"
         run: |
           if [ "${AWS_ACCESS_KEY_ID}" != "" ]; then
+            [ ! -f gib-impacted.log.backup ] || cp gib-impacted.log.backup gib-impacted.log
```

Comment:
looks fragilely easy to forget about

Who could overwrite gib-impacted.log with some incorrect content?

Reply:
This only applies to this job, it's not needed in other cases. Scripts like `plugin/trino-hive-hadoop2/bin/run_hive_alluxio_tests.sh` first check `gib-impacted.log` and then run Maven which overwrites this file, so the check fails when another script like this is called. Maybe we should make a backup in these scripts and restore it after calling Maven before they exit?

Reply:
> then run Maven which overwrites this file, so the check fails when another script like this is called.

why `gib-impacted.log` gets overridden? is it written to implicitly?


> Maybe we should make a backup in these scripts and restore it after calling Maven before they exit?

That's quite error-prone, since I can forget to restore.

Reply:
> > then run Maven which overwrites this file, so the check fails when another script like this is called.
> 
> why `gib-impacted.log` gets overridden? is it written to implicitly?

It's written by Maven but read by a shell script. It get's written every time Maven is executed with GIB enabled and contents depend on the list of modules passed in `-pl`.

> > Maybe we should make a backup in these scripts and restore it after calling Maven before they exit?
> 
> That's quite error-prone, since I can forget to restore.

The user is not expected to do anything, I'm suggesting adding this to the script so it happens automatically.

Reply:
> > why `gib-impacted.log` gets overridden? is it written to implicitly?
> 
> It's written by Maven but read by a shell script. It get's written every time Maven is executed with GIB enabled and contents depend on the list of modules passed in `-pl`.

Can it get written only where we intent it to be? like `mvn -Dwrite-gib-this-time=true`?
is there such a think?

> I'm suggesting adding this to the script so it happens automatically.

you mean the ci.yml? i am concerned it will not get added in all necessary places, because it's more than one, and missing to restore is a silent problem

Reply:
Not `ci.yml` but in all necessary places in:
* plugin/trino-hive-hadoop2/bin/run_hive_adl_tests.sh
* plugin/trino-hive-hadoop2/bin/run_hive_tests.sh
* plugin/trino-hive-hadoop2/bin/run_hive_wasb_tests.sh
* plugin/trino-hive-hadoop2/bin/run_hive_s3_tests.sh
* plugin/trino-hive-hadoop2/bin/run_hive_abfs_access_key_tests.sh
* plugin/trino-hive-hadoop2/bin/run_hive_abfs_oauth_tests.sh
* plugin/trino-hive-hadoop2/bin/run_hive_alluxio_tests.sh

Reply:
> Can it get written only where we intent it to be? like `mvn -Dwrite-gib-this-time=true`? is there such a think?

That's a much better idea, we can set `-Dgib.logImpactedTo=''`

Reply:
Actually i'd prefer to disable writing gib file (eg set `gib.logImpactedTo` to empty by default) 
and enable it only when we explicitly want the gib file to be created.

would that be possible?

Reply:
Done. It's so much better, thank you!